agent/src/workload.rs

```diff
@@ -85,49 +85,51 @@ impl Workload {
         }
     }
     // [impl->swdd~agent-create-control-interface-pipes-per-workload~1]
-    fn set_control_interface(
+    fn exchange_control_interface(
         &mut self,
         pipes_channel_context_info: Option<PipesChannelContextInfo>,
     ) {
         if let Some(control_interface) = self.control_interface.take() {
             control_interface.abort_pipes_channel_task()
         }
         self.control_interface = match pipes_channel_context_info {
-            Some(info) => info.make_context(),
+            Some(info) => info.create_control_interface(),
             None => None,
         };
     }
 
+    fn is_control_interface_changed(
+        &self,
+        pipes_channel_context_info: &Option<PipesChannelContextInfo>,
+    ) -> bool {
+        match (&self.control_interface, pipes_channel_context_info) {
+            (None, None) => false,
+            (Some(_current), None) => true,
+            (None, Some(_new)) => true,
+            (Some(current), Some(new_context)) => {
+                let new_location = new_context.get_workload_instance_name().pipes_folder_name(
+                    pipes_channel_context_info
+                        .as_ref()
+                        .unwrap()
+                        .get_run_folder(),
+                );
+
+                return !current
+                    .get_api_location()
+                    .to_str()
+                    .unwrap()
+                    .contains(new_location.to_str().unwrap());
```

Comment:
Can we use an equal here? I somehow don't have a good feeling with the contains. With all the current implementations, it will work, but it could be a pitfall if we change the ids in a distinct way.
```suggestion
                    .eq(new_location.to_str().unwrap());
```